test/unittests/state_transition.cpp

```diff
@@ -0,0 +1,75 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "state_transition.hpp"
+
+namespace evmone::test
+{
+void state_transition::SetUp()
+{
+    pre.insert(tx.sender, {.nonce = 1, .balance = tx.gas_limit * tx.max_gas_price + tx.value + 1});
+    pre.insert(*tx.to, {.nonce = 1});
+
+    // Default expectations.
+    expect.post[Coinbase].exists = true;
+    expect.post[tx.sender].exists = true;
+    expect.post[*tx.to].exists = true;
+}
+
+void state_transition::TearDown()
+{
+    auto& state = pre;
+    const auto res = evmone::state::transition(state, block, tx, EVMC_CANCUN, vm);
```

Comment:
Would be nice to have revision configurable.

Reply:
Added. For later I'm thinking about revision ranges support.